### DISCUSSION TITLE
No puedo instalar ningún servicio de lenguaje

### DISCUSSION THREAD

**Discussion by QamaQ** [mason-org/mason.nvim#1373] [Q&A]
Hola,  instale Mason.nvim con el instalador de paques Lazy.nvim  y lo hace correctamente, y pues a la hora de ejecutar MasonInstall lua_ls  por ejemplo , no hace ese proceso de instalación, incluso Mason, no muestra ningún servidores de lenguaje

aqui la imagen
![image](https://github.com/williamboman/mason.nvim/assets/117106936/d7a469e4-f538-4bf0-878a-c1f97846ecfe)

y esta es mi configuarcion

```Lua
return {
    "neovim/nvim-lspconfig",

    dependencies = {
        { "folke/neodev.nvim", opts = {} },
        {
            "williamboman/mason.nvim",
            build = ":MasonUpdate",
            opts = {
                ensure_installed = {
                    "stylua",
                    -- "shellcheck",
                    -- "flake8",
                    "shfmt",
                    "mypy",
                    "ruff",
                    "black",
                },
            },
        },
        "williamboman/mason-lspconfig.nvim",
        "jay-babu/mason-null-ls.nvim",
    },
    config = function()
        -- Setup language servers.
        local lspconfig = require("lspconfig")
        local mason = require("mason")
        local mason_lspconfig = require("mason-lspconfig")
        local mason_null_ls = require("mason-null-ls")
        local handlers = require("core.utils")
        mason.setup()
        mason_lspconfig.setup({
            ensure_installed = handlers.language_servers,
            automatic_installation = true,
        })
        mason_null_ls.setup({
            ensure_installed = {
                "prettier",
                "stylua",
                "eslind_d",
                "mypy",
                "ruff",
                "black",
            },
        })

        -- Enable some language servers with the additional completion capabilities offered by nvim-cmp
        local servers = handlers.language_servers

        for _, lsp in ipairs(servers) do
            lspconfig[lsp].setup({
                -- on_attach = my_custom_on_attach,
                --capabilities = handlers.capabilities,
            })
        end

        lspconfig.lua_ls.setup({
            settings = {
                Lua = {
                    completion = {
                        callSnippet = "Replace",
                    },
                    diagnostics = {
                        globals = { "vim" },
                    },
                    telemetry = {
                        enable = false,
                    },
                },
            },
        })

        lspconfig.ltex.setup({
            settings = {
                ltex = {
                    language = { "es", "en" },
                },
            },
        })
    end,
}

```

incluso cuando ejecuto checkhealth mason se queda cargando  y  no hay ningún respuesta,
presioné CTRL+C para que cancele el proceso porque esta bastante tiempo  ahi cargando sin ninguna respuesta

este esl el resultado cunado cancele, después de pasar como 20 min 
![image](https://github.com/williamboman/mason.nvim/assets/117106936/e7ada878-2b76-4892-97ae-e8274adaec05)

**Comment by williamboman** [comment]
Hola! Lo siento para yo solo hablo un poco espanol pero yo entiendo tus problemas. Yo quiero escribir en Espanol tambien para aprender la lingua (yo necesito translate unas pocas palabras 😂).

Yo tengo unas cuantas preguntas:
1) Que "registry settings" tienes para mason.nvim? Es mas facil ver esto cuando presionando `g?` cuando el `:Mason` window esta abierta:
<img width="823" alt="Screenshot 2023-06-28 at 18 03 25" src="https://github.com/williamboman/mason.nvim/assets/6705160/badac21c-d2ff-4523-812d-f2a2097f1317">

---

2) Tienes algunas ERROR/WARN logs en `:MasonLog`? Probablemente tengas network errors que prohibe la descargar el package registry (aunque los errores de descarga deberían mostrarse en el `:Mason` window, lo que me confunde un poco).

> incluso cuando ejecuto checkhealth mason se queda cargando y no hay ningún respuesta, presioné CTRL+C para que cancele el proceso porque esta bastante tiempo ahi cargando sin ninguna respuesta
> 
> este esl el resultado cunado cancele, después de pasar como 20 min

Hm. El health check debe terminar en segundos, no minutos (especialmente no 20 minutos!). Mis suposicion es network errors, aunque errors del network deberian timeout automaticamente despues de un tiempo (~30 segundos).

**Reply by QamaQ** [reply]
Hola, si gracias por responder y me disculpo por la tardanza de la respuesta

![image](https://github.com/williamboman/mason.nvim/assets/117106936/51f34b02-3238-4f6a-a6c1-db9cf50cc366)

El día que compartí este problema, no presioné esas teclas para ver los registro, pero ahora me muestra esto:
![image](https://github.com/williamboman/mason.nvim/assets/117106936/79a9fd48-c1e4-44a2-af97-a8acf0f1260d)

y creo que no hay problema.

![image](https://github.com/williamboman/mason.nvim/assets/117106936/26d7db10-550e-4e5e-94be-492da8c85822)

Si, tenia errores en :MasonLog 
![image](https://github.com/williamboman/mason.nvim/assets/117106936/fb5473f2-2a4f-4930-b304-1f4b703629d6)

y  hasta hoy tengo esos errores en MasonLog, pero Mason puede instalar normalmente y creo no hay problema con eso 😁
![image](https://github.com/williamboman/mason.nvim/assets/117106936/2f717e15-87e2-4fb4-a83b-7a6f048338f2)
